### ISSUE TITLE
Remove HarvestEntityEvent in favor of expansion to DestructEntityEvent.Death

### ISSUE THREAD

**Issue by DarkArc** [SpongePowered/SpongeAPI#971]
At the current time, ````HarvestEntityEvent```` is redundant, and has information that would be useful for player deaths under ````DestructEntityEvent.Death````.

I propose the removal of ````HarvestEntityEvent```` and the expansion of ````DestructEntityEvent.Death```` so as to create a ````DestructEntityEvent.Death.Player```` which has the currently available methods of ````HarvestEntityEvent.TargetPlayer````.

**Comment by gabizou** [comment]
You can have multiple cases of `HarvestEntityEvent` including when you shear sheep (as an example), so the event cannot just be bundled in with death.

**Comment by DarkArc** [comment]
The information from sheering a sheep can however be derived from DropItemEvent and InteractEntityEvent.

**Comment by ST-DDT** [comment]
For reference: [HarvestEntityEvent](https://github.com/SpongePowered/SpongeAPI/blob/master/src/main/java/org/spongepowered/api/event/entity/HarvestEntityEvent.java#L37-L38)

> Called when an `Entity` has been killed and is being "harvested" (drops/etc). Happens after `DestructEntityEvent`.

@gabizou But then javadocs are wrong, since sheering usually does not kill the sheep.

**Comment by Zidane** [comment]
@DarkArc 

I slept on this last night and things have changed since we did the last big event refactor. We can now track why the sheep got sheared (aka a Player did it) in the contextual data of DropItemEvent (Sheep would be the immediate cause, Player would be contextual). If there are no items dropped (aka a mod puts something into your inventory from an Entity), then that would trigger an appropriate inventory event again with cause/contextual data.

In short, this can just be removed.

**Comment by DarkArc** [comment]
@Zidane would these item drop events be able to prevent the items from leaving the inventory when a player dies, or prevent their XP from being lost?

**Comment by Zerthick** [comment]
I discussed this a bit in #938, in short it would be useful to have an event that controls all the drops from an entities death (items & exp), that way if, for instance, you wanted to prevent all drops you could just listen to, and cancel one event rather than multiple events.  Perhaps that could be the purpose of ```HarvestEntityEvent```?
